nameable.rb

```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses must implement this method'
+  end
+end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  MAX_LENGTH = 10
+
+  def correct_name
+    name = @nameable.correct_name
+    name.length > MAX_LENGTH ? name[0...MAX_LENGTH] : name
+  end
```

Comment:
- [ ] [OPTIONAL] It would have been better if these classes were in separate folders for readability purposes.

Reply:
Okay. Thanks, @elisha2kyakpo1. I will carry out the suggestion immediately.